src/Elastic.Apm/Api/IMetricSet.cs

```diff
@@ -0,0 +1,20 @@
+using System.Collections.Generic;
+
+namespace Elastic.Apm.Api
+{
+	/// <summary>
+	/// Data captured by the agent representing a metric occurring in a monitored service
+	/// </summary>
+	public interface IMetricSet
```

Comment:
This is just to follow the convention of using interfaces in the `IPayloadSender` method. In case we e.g. implement Noop transactions, it’ll be very useful to have e.g. `ITransaction` on `IPayloadSender`, so we can pass a different implementation to the `IPayloadSender`. Maybe we end up with something similar for metrics. If not I think it’s still more unified to only accept interfaces on `IPayloadSender` and not implementations.